### ISSUE TITLE
Autopep8 adds a space before every @ decorator when syntax is incorrect

### ISSUE THREAD

**Issue by NicolasM-Forsk** [hhatto/autopep8#666]
When python syntax is incorrect (ex: unclosed parenthesis), autopep8 adds a space after "@".
It doesn't remove it when I fix the code syntax.

Could you please improve that?
Thanks a lot

---

## Python Code
```python

def toto():
    print(


def tata():
    pass


@staticmethod
def decorated_function():
    pass


@staticmethod
def decorated_function2():
    pass
```

## Command Line and Configuration
default

Command Line
default

## Your Environment
* VS Code with format on save
* Python version: Python 3.8.10
* autopep8 version : 2.0.0 (pycodestyle: 2.10.0)
* Platform: ubuntu 20.04

**Comment by hhatto** [comment]
If an invalid code is entered, pycodestyle/autopep8 does not support the correct behavior.

**Comment by nmoreaud** [comment]
So, you think it's ok to have the whole file containing:
```python
@ staticmethod
def decorated_function():
    pass


@ staticmethod
def decorated_function2():
    pass
```

There doesn't seem to exist a PEP 8 rule concerning this space, though I've never seen a code on the internet with a space between `@` and the decorator.
Can I disable autopep8 when the code is invalid?

**Comment by antoine-sac** [comment]
The problem is not so much that it introduces the space when the code is incorrect, but that it doesn't remove the space once the code is correct.

So the file in question is left with spaces in all decorators even once the code is correct.

Shouldn't autopep8 remove the unexpected spaces that it introduced?

**Comment by antoine-sac** [comment]
My bad, this is fixed in v2.3, I just had to upgrade autopep8.

**Comment by hhatto** [comment]
fix this issue in v2.3.0+
https://github.com/hhatto/autopep8/releases/tag/v2.3.0
